libbeat/beat/beat.go

```diff
@@ -607,7 +608,27 @@ func (b *Beat) registerTemplateLoading() error {
 
 	// Loads template by default if esOutput is enabled
 	if b.Config.Output.Name() == "elasticsearch" {
+		var cfg template.TemplateConfig
+		err := b.Config.Template.Unpack(&cfg)
+		if err != nil {
+			return fmt.Errorf("unpacking template config fails: %v", err)
+		}
+
+		// Get ES Index name for comparison
+		esCfg := struct {
+			Index string `config:"index"`
+		}{}
+		err = b.Config.Output.Config().Unpack(&esCfg)
+		if err != nil {
+			return err
+		}
+
+		if esCfg.Index != "" && (cfg.Name == "" || cfg.Pattern == "") {
+			return fmt.Errorf("setup.template.name and setup.template.pattern have to be set if index name is modified.")
```

Comment:
It will be kind of strange if the name is not changed. Assuming someone changes the index to `test-{daily}` and the pattern to `test-*` he then sees `beatname-{version}-{daily}` as the template name with `test-*` as pattern. In case that template is already there, it would not be overwritten and he would still not have the expected template.